File: src/Marvin.Runtime.Maintenance.Web.UI/src/modules/models/ModuleServerModuleState.ts
/*
 * Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
 * Licensed under the Apache License, Version 2.0
*/

export enum ModuleServerModuleState {
    Stopped =  0x0,
    Initializing = 0x2,
    Ready = 0x1,
    Starting = 0x3,
    Running = 0x8,
    Stopping = 0xA,
    Failure = 0x4,
}

File: src/Moryx.Runtime.Maintenance/Maintenance/Modules/Models/ModuleServerModuleState.cs
// Copyright (c) 2020, Phoenix Contact GmbH & Co. KG
// Licensed under the Apache License, Version 2.0

namespace Moryx.Runtime.Maintenance.Modules
{
    /// <summary>
    /// Server module state
    /// </summary>
    public enum ModuleServerModuleState
    {
        /// <summary>
        /// Initial value
        /// </summary>
        Stopped,

        /// <summary>
        /// Module is initializing
        /// </summary>
        Initializing,

        /// <summary>
        /// Service is ready to be started
        /// </summary>
        Ready,

        /// <summary>
        /// Service is starting
        /// </summary>
        Starting,

        /// <summary>
        /// Service is running
        /// </summary>
        Running,

        /// <summary>
        /// Service is stopping
        /// </summary>
        Stopping,

        /// <summary>
        /// Service failed
        /// </summary>
        Failure,
    }
}
